tools/introspection.h

```diff
@@ -142,6 +142,12 @@ struct ulp_addresses
   Elf64_Addr global;
 };
 
+void release_ulp_thread(struct ulp_thread *);
+
+void release_ulp_dynobj(struct ulp_dynobj *);
+
```

Comment:
You seem to call only release_ulp_process from the outside code, so please only export that one and make the _thread and _dynob helpers static and internal.